src/sentry/grouping/strategies/newstyle.py

```diff
@@ -733,6 +740,83 @@ def chained_exception(
     return rv
 
 
+# See https://github.com/getsentry/rfcs/blob/main/text/0079-exception-groups.md#sentry-issue-grouping
+def filter_exceptions_for_exception_groups(
+    exceptions: List[SingleException], context: GroupingContext, event: Event, **meta: Any
+) -> List[SingleException]:
+
+    # This function only filters exceptions if there are at least two exceptions.
+    if len(exceptions) <= 1:
+        return exceptions
+
+    # Reconstruct the tree of exceptions if the required data is present.
+    nodes = {}
+    for exception in reversed(exceptions):
+        exception._children = []
+        mechanism: Mechanism = exception.mechanism
+        if mechanism and mechanism.exception_id is not None:
+            nodes[mechanism.exception_id] = exception
+            if mechanism.parent_id is not None:
```

Comment:
In theory that shouldn't happen because the outer for-loop is reversed.  Since it's reconstructing the tree starting with the root node, the parent node will always be created before the child node is added to it.

However, that assumes the SDK sent the data correctly.  We could be more resilient to garbage data here.

Reply:
> However, that assumes the SDK sent the data correctly. We could be more resilient to garbage data here.

if you crash here, the event is dropped entirely.

Reply:
FWIW I'm not only talking about order here, I was more thinking of the case where the exception pointed to by parent_id does not exist at all

Reply:
I resolved this by not requiring that the parent nodes be pre-existing in the `exception_tree` dictionary.  I also do a try/except around the entire function, so there's no longer a risk that the event will be dropped.  Worst case, all exceptions will be used for grouping instead of a filtered subset.